contrib/zip/utils/utils_zip.h

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CONTRIB_ZIP_UTILS_UTILS_ZIP_H_
+#define CONTRIB_ZIP_UTILS_UTILS_ZIP_H_
+
+#include <fcntl.h>
+
+#include "contrib/zip/sandboxed.h"
+#include "sandboxed_api/util/status_macros.h"
+
+class LibZip {
+ public:
+  explicit LibZip(ZipSandbox* sandbox, std::string filename, int flags)
+      : sandbox_(CHECK_NOTNULL(sandbox)),
+        api_(sandbox_),
+        filename_(std::move(filename)),
+        flags_(flags),
+        rfd_(open(filename.c_str(), O_RDWR | O_CREAT)) {
+    OpenRemote().IgnoreError();
+  };
+  ~LibZip();
+
+  bool IsOpen();
+
+  absl::StatusOr<std::string> GetName(uint64_t i);
+  absl::StatusOr<uint64_t> GetNumberEntries();
+  absl::StatusOr<std::vector<uint8_t>> ReadFile(uint64_t index);
+  absl::StatusOr<std::vector<uint8_t>> ReadFile(std::string filename);
+  absl::StatusOr<uint64_t> AddFile(std::string filename,
+                                   std::vector<uint8_t>& buf);
+  absl::StatusOr<uint64_t> AddFile(std::string filename, int fd);
+  absl::Status ReplaceFile(uint64_t index, std::vector<uint8_t>& buf);
+  absl::Status ReplaceFile(uint64_t index, int fd);
+  absl::Status DeleteFile(uint64_t index);
+
+  absl::StatusOr<std::string> GetError();
+
+  absl::Status Finish();
```

Comment:
I think `Finish()` should have a comment wrt the class invariants. For example, what happens when I call `AddFile()` (or anything else that modifies the ZIP) after `Finish()`?

Reply:
All my function has invariants.
At the begging of each function we check:
```
SAPI_RETURN_IF_ERROR(CheckOpen());
```

or
```
SAPI_RETURN_IF_ERROR(CheckClosed());
```

I will change Closed to Finished.